### PR TITLE
Create 0783-minimum-distance-between-bst-nodes.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved minimum-distance-between-bst-nodes in JS.

File(s) Added: 0783-minimum-distance-between-bst-nodes.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/minimum-distance-between-bst-nodes/submissions/1346793747/

### CODE REVIEW

File: javascript/0783-minimum-distance-between-bst-nodes.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * https://leetcode.com/problems/minimum-distance-between-bst-nodes/
+ * @param {TreeNode} root
+ * @return {number}
+ */
+var minDiffInBST = function(root) {
+    
+    const sortedArr = [];
+
+    // levelOrderTraversal
+    const dfs = (node) => {
```
Comment (aakhtar3) [inline javascript/0783-minimum-distance-between-bst-nodes.js:21]: Avoid nestled functions
```js

/**
 * Iterative - BST
 * Time O(N) | Space O(N)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = (root, minDistance = Infinity) => {
    const inorderNodes = inorderTraversal(root);

    for (let i = 1; (i < inorderNodes.length); i++) {
        minDistance = Math.min(minDistance, (inorderNodes[i] - inorderNodes[(i - 1)]));
    }
    
    return minDistance;
}

var inorderTraversal = (root, inorderNodes = []) => {
    const isBaseCase = (root === null);
    if (isBaseCase) return inorderNodes;
    
    inorderTraversal(root.left, inorderNodes);
    inorderNodes.push(root.val);
    inorderTraversal(root.right, inorderNodes);

    return inorderNodes;
}

/**
 * DFS - BST
 * Time O(N) | Space O(N)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = (root, prev = [ null ], minDistance = [ Infinity ]) => {
    const isBaseCase = (root === null);
    if (isBaseCase) return minDistance[0];

    minDiffInBST(root.left, prev, minDistance);

    const canUpdate = (prev[0] !== null);
    if (canUpdate) minDistance[0] = Math.min(minDistance[0], (root.val - prev[0].val));

    prev[0] = root;

    minDiffInBST(root.right, prev, minDistance);

    return minDistance[0];
}
```

Comment (aadil42) [reply javascript/0783-minimum-distance-between-bst-nodes.js:21]: Omitting nested funciton.
Submission link for the proposed code: https://leetcode.com/problems/minimum-distance-between-bst-nodes/submissions/1388701576/

```javascript
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * DFS (level order traversal)
 * Time O(n) | Space O(n)
 * https://leetcode.com/problems/minimum-distance-between-bst-nodes/
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function(root) {
    // levelOrderTraversal
    const sortedArr = dfs(root, []);
    
    let min = Infinity;
    for(let i = 1; i < sortedArr.length; i++) {
        min = Math.min(min, sortedArr[i] - sortedArr[i-1]);
    }
    return min;
};

const dfs = (node, sortedArr) => {
    if(!node) return;

    dfs(node.left, sortedArr);
    sortedArr.push(node.val)
    dfs(node.right, sortedArr);

    return sortedArr;
}
```

Comment (aadil42) [reply javascript/0783-minimum-distance-between-bst-nodes.js:21]: @aakhtar3 (tagging to draw attention to the proposed code)

Comment (Ykhan799) [reply javascript/0783-minimum-distance-between-bst-nodes.js]: @aadil42 For line 22, add spacing for the i - 1. Other than that the rest of the solution looks good 👍

Comment (aadil42) [reply javascript/0783-minimum-distance-between-bst-nodes.js]: @Ykhan799, it's done.

**Pull request merged**